### DISCUSSION TITLE
Questions about bounties around improving blog, support pages & UI/UX

### DISCUSSION THREAD

**Discussion by thinkinginbinary** [simple-login/app#1740] [Q&A] [answered]
Hello,  I am an engineer who is interested in bounties for security improvements, UI/UX improvements. & information consistency improvement regarding SL. 

I am wondering if email is the best channel or if it is ok to submit security reports or concepts around UI/UX improvements pubicly via GH discussions? 

Any thoughts or ideas are welcome. I look forward to comversating about this more.

**Comment by fm** [comment]
Being an open source project, anyone can contribute PRs/improvements, but you will not be paid for them and it's up to the code owners to accept, deny or request further changes. 

Security vulnerabilities should be reported to Proton via their responsible disclosure program and may be paid out according to [Proton's bounty program ](https://proton.me/blog/protonmail-bug-bounty-program)

**Reply by thinkinginbinary** [reply]
@fm  Why do SL posts indicate being rewarded to improve SL blog, support pages via the bounty program?

**Comment by t2noob** [comment]
Here is a bit of information on the Simplelogin bounty program
https://www.reddit.com/r/ProtonMail/comments/y6au4h/simplelogin_is_now_included_in_our_bug_bounty/

**Reply by thinkinginbinary** [reply]
Thanks for this,  there are three different emails listed across different posts & a mention of an online site bugbounty.ch

**Comment by thinkinginbinary** [comment]
Ok thank you. I see a different email in this post. The PM security team's
origonal response was not thorough. I have already reached out to them.
Does SL not have anyone working on security or managing bounty related
rewards internally?


Sincerely,
Matt Shaver

**Reply by fm** [reply, marked as answer]
SL was only a handful of people before they joined Proton. As they are now part of Proton, their security teams are unified. If you find any issues you can either submit a PR with a fix, or submit a report using the links in previous posts.

**Reply by thinkinginbinary** [reply]
Ok do you happen to have any info about bounties for SL site improvements? The security team stated that bounties are only provided for vulnerabilities. SL post(s) state bounties can be provided for site improvements. I created a report showing how current services can be used for attacks, & asked about abuse protection mechanisms clearly stating that I was sharing a vulnerability. In prior emails I was told that bounties are only provided for security vulnerabilities. It has been about 1 week since I last sent an email & to the best of my current knowledge no one has responded. This may be due to the wording focusing on intentional design choices,intention does not null out attack possibilities or invalidate information. It would be great to get more insight.  

If a bit more work is needed somewhere that is understandable. Although obviously working for no non xp gain is hopefully not an expectation as so far my experience with bounty programs is showing that there may be an unrealistic expectation from companies that researchers should work solely for experience until bounties are rewarded. It really brings up the question of whether bounty programs are ethical ways to improve security. Ironically social data seems to indicate that people think it is unethical to earn a decent income from sharing research. If research leads to job opportunities that is great, unpaid internships especially with no clear reward outside of xp gained are now considered unethical basically everywhere & so why should bounty programs be acceptable in this way? Woludn't it make more sense to have a minimum financial reward for all participants who make clear efforts regardless of outcomes or expectations? How do bounty programs that don't reward participants not violate labor laws as participation from researchers with no financial gain is basically equating to completely free labor for the company?

Just want to share these thoughts as maybe they can be used to improve current programs or leveraged for future class actions against platforms with miserable business designs as that is likely needed to improve tech security & agencies arguing for poor security isn't logical(it seems clear this is happening due to rampant poor security across many tech companies) as it opens the door for non agency actors to take advantage of the poor security as there is no reason that agencies with vast resources should not surpass improved security collectively anyway. Increased security also helps mitigate corruption due to higher effort needed from individual actors which basically means that collective agreement is a neccesity & non optional at some point. Though obviously work effort & time are a big deal also so maybe that is considered the logical part from some perspectives.

**Reply by fm** [reply]
It is quite likely that the SL posts are deprecated now and the Proton policy is the reigning policy. I am not an SL/Proton employee - so I can't help you with what they decide to do with your report.

**Reply by thinkinginbinary** [reply]
Maybe so. I find that doubtful due to the page(s) remaining as it would require intentional deciaion making around depreciation. True. I am juat opening a discussion, hopefully some employees will respond.

**Reply by nguyenkims** [reply]
Yes I can confirm that the Proton bug bounty program should be used here instead of the past SimpleLogin program. Where do you see the past program so we can update it?

**Reply by thinkinginbinary** [reply]
@nguyenkims 

https://simplelogin.io/blog/bug-bounty-program/ - Qualifying improvements is really vague & leads to questions such as can I be rewarded for finding these improvements? 

https://simplelogin.io/pricing/ - States nothing about free trials.

https://simplelogin.io/contact/ - This page lists a different email than the one on the bounty program page.

https://proton.me/blog/protonmail-bug-bounty-program - This page also lists the qualifying improvements section, which states "information disclosure" as a potential rewarded improvement area, clairty about what path/information disclosure means would be great. 

https://proton.me/blog/bug-bounty-partnership-continues - This page does not align with other pages at all as it lists a 3rd party company.

https://proton.me/security/response-center - No threat model for Proton VPN or SL.is listed which may also lead to confusion around whether SL, VPN issues should lead to contacting the PM security team. 

https://www.bugbounty.ch/en/proton - This program doesn't mention SL & leads to lack of clarity due to there being two completely different approaches to bounties. It also lists ProtonVPN & thos ProtonVPN page does not list this site at all https://protonvpn.com/blog/bug-bounty-program/ (In the comments you can see how others are also not agreeing with the minimum financial rewards for program participation, which can even technically be $0 if all researcher reports are denied, how that is even legal is beyond my comprehension. Some comments are entirely not responded to which leads to distrust in PM team from users.) 

https://proton.me/blog/bug-bounty-pr - This page also lists bugbounty.ch. Out of the
two options  of contacting PM security tram or uaing thw programme as a researcher I have no idea which one is preferred & when looking at the most easily found site pages if the bugbounty.ch programme is actively being maintained. The max rewards are much higher on the bugbounty.ch programme, there is no info as to why the max rewards appear to be 3x higher.

I think there should be much improved minimum rewards for people who participate in the bug bounty program as this would lead to higher quality work, & labor should not be free to PM, SL teams even if the results are not always meeting all specific expectation(s). Some metrics or processes for evaluating a minimum level of researcher effort should be established vs expecting  or wanting a specific type of vulnerability. 

The report I shared previously also should lead to improved SL DOS safeguards if analysed fully which does fall under qualifying improvements.. I won't post the report here due to the programme ToS listing not to disclose reports pubicly until issues are resolved. 

There also this line which my report, improvements should fall under on the PM bug bounty page: Any design or implementation issue that substantially affects the confidentiality or integrity of user data is likely to be in scope for the program. At scale that what I mentioned in my report could compromise the integrity of user data by causing server failure, resource exhaustion, increased  server visilbility due to abuse & possible server black listing(based on my current understanding of how the SL service works, as it seems to mainly be a middle man mail server more or less.) which may make the server(s) a target for external investigative attacks. 

Another issue I ran into is that when using reverse aliases even after setting options to remove sender address the recepient name is still showing next to the reverse aliased contact.(This was with PGP keys configured.). I have not tested that on lots of accounts. 

I hope my efforts help improve SL & PM services, products. I would also really like to know if it is possible to get access to internal testing tools, any resource around abuse protection mechanisms & to be rewarded for time, effort so far somehow. I am happy to clarify any information on the report I submitted & according to the bounty program design decisions are still included also.

**Reply by thinkinginbinary** [reply]
Btw apologies for the late response.

**Reply by nguyenkims** [reply]
Hi as SimpleLogin is now in Proton, we use Proton bounty program indeed that you can find more detials on https://proton.me/security/bug-bounty

This program is limited to code though so feedbacks on the copy isn't included ...

I'd suggest to contact security@proton.me for the issue related to DDOS and reverse alias you've found, our security team will exchange with you on the details and the rewards.

**Comment by thinkinginbinary** [comment]
Thank you for the clarification. I sent another email today that hopefully
leads to more productive communications.

Would a PR without a detailed code resolution be accepted?

**Comment by thinkinginbinary** [comment]
Are there pages that indicate communication timeframes for different inquiry types?
